public/src/main/java/com/revenuecat/purchases/Offering.kt

```diff
@@ -15,6 +15,7 @@ package com.revenuecat.purchases
 data class Offering constructor(
     val identifier: String,
     val serverDescription: String,
+    val metadata: Map<String, Any>,
```

Comment:
Mentioned this in the iOS PR, but I wonder if this should be a custom class with accessors for the data that force developers to provide default values.

Reply:
I'm trying to do something similar to what we did in Swift and... its not working 🥴

```kotlin
fun <T> getMetadataValue(key: String, default: T) : T {
    return try {
        @Suppress("UNCHECKED_CAST")
        val value = metadata[key] as? T
        value ?: default
    } catch (e: ClassCastException) {
        default
    }
}
```

I don't think `val value = metadata[key] as? T` works the same way as it does in Swift because `metadata[key]` is `Any?` and its now downcasting 😬 

```kotlin
val wrongMetadataType = offering.getMetadataValue("string", 5.5)
```

This example above crashes because `getMetadataValue()` wants to return a `Double` and is expecting a `Double` but its returning a `String` (because Kotlin isn't downcasting).

Is this same API style not possible here? 😬  (cc: @NachoSoto)